code-reviews/recipes/CSharp.md

```diff
@@ -6,15 +6,80 @@
+
+Details about the configuration of different rules can be found [here](https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2019).
+
+### FxCop analyzers
+
+Microsoft's FxCop analyzers check your code for security, performance, and design issues, among others. [Install FxCop analyzers in Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/install-fxcop-analyzers?view=vs-2019).
+
+You can install the FxCop analyzers using nuget or a VSIX extension. We reocommend using the nuget package ([see Project setup](#project-setup)). Which allows for consistent use accross all developers on a project as well as CI validation.
+
+### StyleCop analyzer
+
+The StyleCop analyzer is a nuget package (StyleCop.Analyzers) that can be installed in any of your projects. It's mainly around code style rules and makes sure, the team is following the same rules without having subjective discussions around braces and spaces. Detailed information can be found here: [StyleCop Analyzers for the .NET Compiler Platform](https://github.com/DotNetAnalyzers/StyleCopAnalyzers).
+
+The minimum rules set teams should adopt is the [Managed Recommended Rules](https://msdn.microsoft.com/en-us/library/dd264893.aspx) rule set.
+
+## Automatic Code Formatting
+
+Use .editorconfig to configure code formatting rules in your poject.
+
+## Build validation
+
+It's important that you enforce your code style and rules in the CI to avoid any team member merging code that does not comply with your standards into your git repo.
+
+If you are using FxCop analyzers and StyleCop analyzer, it's very simple to enable those in the CI. You have to make sure you are setting up the project using nuget and .editorconfig ([see Project setup](#project-setup)). Once you have this setup, you will have to configure the pipeline to build your code. That's pretty much it. The FxCop analyzers will run and report the result in your build pipeline. If there are rules that are violated, your build will be red.
+
+```yaml
+    - task: DotNetCoreCLI@2
+      displayName: 'Style Check & Build'
+      inputs:
+        command: 'build'
+        projects: '**/*.csproj'
+```
 
 ## Code Review Checklist
 
-* [ ] Is framework version of code compatible with the target system
-* [ ] Does this code make correct use of [asynchronous programming constructs](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/#BKMK_AsyncandAwait), including proper use of ```await``` and ```Task.WhenAll```?
+* [ ] Does this code make correct use of [asynchronous programming constructs](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/#BKMK_AsyncandAwait), including proper use of ```await``` and ```Task.WhenAll``` including CancellationTokens?
+* [ ] Is the code subject to concurrency issues? Are shared objects property
```

Comment:
What is the meaning of the last sentence?
- Are shared objects property-protected?
- Are shared objects properly protected?
- Are shared objects properties protected?

Reply:
properly protected. :) Thanks!